lib/Pass/Pass.cpp

```diff
@@ -386,102 +386,109 @@ struct SouperPass : public ModulePass {
         }
         AddToCandidateMap(CandMap, R);
       }
+    }
 
 
-      for (auto &Cand : CandMap) {
+    for (auto &Cand : CandMap) {
```

Comment:
After the block information is attached to the LHS, the for loop in Pass.cpp now gets reverted. Now it first collects all the candidates at one time, instead of collects the candidate by each block.